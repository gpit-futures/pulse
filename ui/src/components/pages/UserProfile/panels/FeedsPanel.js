import React, { PureComponent } from 'react';

import ControlPanel from '../ControlPanel';
import Feeds from '../../Feeds/Feeds'

const FEEDS = 'feeds';

export default class FeedsPanel extends PureComponent {
  render() {
    const {
      openedPanel,
      editedPanel,
      onShow,
      onExpand,
      onEdit,
      onCancel,
      isShowControlPanel,
      isSaveButton,
    } = this.props;

    return (
      <ControlPanel
        name={FEEDS}
        title="Feeds"
        isOpen={openedPanel === FEEDS}
        onShow={onShow}
        onExpand={onExpand}
        onEdit={onEdit}
        editedPanel={editedPanel}
        onCancel={onCancel}
        isShowControlPanel={isShowControlPanel}
        isSaveButton={isSaveButton}
      >
        <Feeds />
      </ControlPanel>
    )
  }
}