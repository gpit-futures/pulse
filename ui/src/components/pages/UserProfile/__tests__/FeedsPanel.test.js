import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';

import FeedsPanel from '../panels/FeedsPanel';

Enzyme.configure({ adapter: new Adapter() });

const FEEDS = 'feeds';

const testProps = {
  openedPanel: FEEDS,
  editedPanel: {},
  onShow: () => {},
  onExpand: () => {},
  onEdit: () => {},
  onCancel: () => {},
  isShowControlPanel: false,
  isSaveButton: false,
};

describe('Component <FeedsPanel />', () => {
  it('should renders with all props correctly', () => {
    const component = shallow(
      <FeedsPanel
        openedPanel={testProps.openedPanel}
        editedPanel={testProps.editedPanel}
        onShow={testProps.onShow}
        onExpand={testProps.onExpand}
        onEdit={testProps.onEdit}
        onCancel={testProps.onCancel}
        isShowControlPanel={testProps.isShowControlPanel}
        isSaveButton={testProps.isSaveButton}
      />
    );

    expect(component.find('ControlPanel')).toHaveLength(1);
    expect(component.find('ControlPanel').props().title).toEqual('Feeds');
    expect(component.find('ControlPanel').props().name).toEqual('feeds');

    expect(component).toMatchSnapshot();

  });
});

