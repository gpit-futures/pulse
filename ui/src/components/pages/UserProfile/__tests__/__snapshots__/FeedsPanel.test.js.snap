// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <FeedsPanel /> should renders with all props correctly 1`] = `
<ControlPanel
  editedPanel={Object {}}
  isOpen={true}
  isSaveButton={false}
  isShowControlPanel={false}
  name="feeds"
  onCancel={[Function]}
  onEdit={[Function]}
  onExpand={[Function]}
  onShow={[Function]}
  title="Feeds"
>
  <Connect(Connect(Connect(Connect(lifecycle(lifecycle(Feeds)))))) />
</ControlPanel>
`;
